algopy_testing/src/algopy/utils.py

```diff
@@ -59,3 +59,19 @@ def as_bytes(value: object, *, max_size: int = MAX_BYTES_SIZE) -> bytes:
     if len(bytes_value) > max_size:
         raise ValueError(f"expected value length <= {max_size}, got: {len(bytes_value)}")
     return bytes_value
+
+
+def as_string(value: object, *, max_size: int = MAX_STRING_SIZE) -> str:
+    from algopy import String
+
+    match value:
+        case str(string_value):
+            pass
+        case String(value=string_value):
+            pass
+        case _:
+            raise TypeError(f"value must be a string or String type, not {type(value).__name__!r}")
+    if len(string_value) > MAX_STRING_SIZE:
```

Comment:
Since UTF8 can contain code points consisting of multiple bytes, we can't check the string length, instead need to check the length of the underlying bytes e.g.
```suggestion
    if len(string_value.encode("utf8")) > MAX_BYTES_SIZE:
```